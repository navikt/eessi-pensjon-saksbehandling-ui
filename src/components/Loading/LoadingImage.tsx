import { progressBarAnimation } from '@navikt/hoykontrast'
import styled from 'styled-components/macro'

export const ImageDiv = styled.div`
  min-width: 40px;
  min-height: 40px;
  background-color: whitesmoke;
  border-radius: 100px;
  transition: width .6s ease;
  animation: ${progressBarAnimation} 1s linear infinite;
  background-image: linear-gradient(45deg, hsla(90, 90%, 90%, .15) 25%, transparent 0, transparent 50%, hsla(90, 90%, 90%, .15) 0, hsla(90, 90%, 90%, .15) 75%, transparent 0, transparent);
`
const LoadingImage = (props: any) => (<ImageDiv {...props} />)

export default LoadingImage
