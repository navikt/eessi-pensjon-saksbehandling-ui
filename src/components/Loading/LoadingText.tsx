import { progressBarAnimation } from '@navikt/hoykontrast'
import styled from 'styled-components'

export const TextDiv = styled.div`
  min-width: 50%;
  transition: width .6s ease;
  background-color: whitesmoke;
  animation: ${progressBarAnimation} 1s linear infinite;
  background-image: linear-gradient(45deg, hsla(90, 90%, 90%, .15) 25%, transparent 0, transparent 50%, hsla(90, 90%, 90%, .15) 0, hsla(90, 90%, 90%, .15) 75%, transparent 0, transparent);
`

const LoadingText = (props: any) => (<TextDiv {...props} />)

export default LoadingText
