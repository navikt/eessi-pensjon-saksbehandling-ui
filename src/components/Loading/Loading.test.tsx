import LoadingImage, { ImageDiv } from 'components/Loading/LoadingImage'
import LoadingText, { TextDiv } from 'components/Loading/LoadingText'
import { render } from '@testing-library/react'

describe('components/Loading/LoadingImage', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = render(<LoadingImage />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    const { container } = render(<LoadingImage />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.find(ImageDiv)).toBeTruthy()
  })
})

describe('components/Loading/LoadingText', () => {
  let wrapper: any

  beforeEach(() => {
    wrapper = render(<LoadingText />)
  })

  afterEach(() => {
    wrapper.unmount()
  })

  it('Render: match snapshot', () => {
    const { container } = render(<LoadingText />)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Render: has proper HTML structure', () => {
    expect(wrapper.find(TextDiv)).toBeTruthy()
  })
})
