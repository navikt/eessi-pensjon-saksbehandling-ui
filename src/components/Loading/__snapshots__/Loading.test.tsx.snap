// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Loading/LoadingImage Render: match snapshot 1`] = `
.c1 {
  min-width: 40px;
  min-height: 40px;
  background-color: whitesmoke;
  border-radius: 100px;
  -webkit-transition: width .6s ease;
  transition: width .6s ease;
  -webkit-animation: da-daAO 1s linear infinite;
  animation: da-daAO 1s linear infinite;
  background-image: linear-gradient(45deg,hsla(90,90%,90%,.15) 25%,transparent 0,transparent 50%,hsla(90,90%,90%,.15) 0,hsla(90,90%,90%,.15) 75%,transparent 0,transparent);
}

ui-fss.g48[id="c0"] {
  content: "c1,";
}

<LoadingImage>
  <styled.div>
    <div
      className="c0 c1"
    />
  </styled.div>
</LoadingImage>
`;

exports[`components/Loading/LoadingText Render: match snapshot 1`] = `
.c1 {
  min-width: 40px;
  min-height: 40px;
  background-color: whitesmoke;
  border-radius: 100px;
  -webkit-transition: width .6s ease;
  transition: width .6s ease;
  -webkit-animation: da-daAO 1s linear infinite;
  animation: da-daAO 1s linear infinite;
  background-image: linear-gradient(45deg,hsla(90,90%,90%,.15) 25%,transparent 0,transparent 50%,hsla(90,90%,90%,.15) 0,hsla(90,90%,90%,.15) 75%,transparent 0,transparent);
}

ui-fss.g48[id="c0"] {
  content: "c1,";
}

<LoadingImage>
  <styled.div>
    <div
      className="c0 c1"
    />
  </styled.div>
</LoadingImage>
`;
